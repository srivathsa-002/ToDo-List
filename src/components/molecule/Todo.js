import React from "react";
import "../../styles/components/option.css";
import EditIcon from "../atoms/EditIcon";
import DeleteIcon from "../atoms/DeleteIcon";
import CheckboxIcon from "../atoms/CheckboxIcon";
import UpdateIcon from "../atoms/UpdateIcon";
import Para from "../atoms/Para";

export default class Todo extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isEditing: false
    }
    this.renderForm = this.renderForm.bind(this);
    this.renderTodos = this.renderTodos.bind(this);
    this.toggleState = this.toggleState.bind(this);
    this.updateTodo = this.updateTodo.bind(this);
  }

  updateTodo(event) {
    event.preventDefault();
    this.props.editTodo(this.props.todo, this.input.value);
    this.toggleState();
  }

  toggleState() {
    const { isEditing } = this.state;
    this.setState({
      isEditing: !isEditing
    })
  }

  handleNewVal(newVal) {
    console.log(newVal);
    this.updateTodo();
  }

  renderForm() {
    return (
      <form className="option" onSubmit={this.updateTodo} data-testid="form">
        <input 
          type="text" 
          ref={(val) => {
            this.input = val
          }} 
          defaultValue={this.props.todo.text} 
        />
        <UpdateIcon placement="bottom" click={this.updateTodo} />
        
      </form>
    )
  }

  renderTodos() {
    return (
      <div className="option" data-testid="todo">
        <div className="option-text">
          <Para val={this.props.todo.text}/>
        </div>
        <div className="option-materials">
          <EditIcon placement="bottom" click={() => this.toggleState()} />
          
          <CheckboxIcon placement="bottom" checked={this.props.todo.complete ? true : false} click={this.props.toggleComplete} />
          
          <DeleteIcon placement="bottom" click={this.props.onDelete} />
          
        </div>
      </div>
    )
  }

  render() {
    const { isEditing } = this.state;
    return (
      <section>
        {
          isEditing ? this.renderForm() : this.renderTodos()
        }
      </section>
    );
  }
}

/* <Tooltip title="Delete ToDo" placement="bottom" arrow>
            <div>
              <IconButton>
                <DeleteOutlineOutlinedIcon onClick={this.props.onDelete} />
              </IconButton>
            </div>
          </Tooltip> */

/* <Tooltip title="Mark as Complete" placement="bottom" arrow>
            <div>
              <IconButton>
                <CheckBox isChecked={this.props.todo.complete ? true : false} onClick={this.props.toggleComplete} />
              </IconButton>
            </div>
          </Tooltip> */

/*<Tooltip title="Edit ToDo" placement="bottom" arrow>
            <div>
              <IconButton>
                <CreateOutlinedIcon onClick={() => this.toggleState()} />
              </IconButton>
            </div>
          </Tooltip> */

/* <p style={{ margin: "0" }}>{this.props.todo.text}</p> */

/* <Tooltip title="Update Todo" placement="bottom" arrow>
          <div>
            <IconButton onClick={this.updateTodo}>
              <ArrowUpwardIcon />
            </IconButton>
          </div>
        </Tooltip> */

/* <UpdateInputBox
          onNewVal={this.handleNewVal}
          defaultValue={this.props.todo.text}
        /> */

