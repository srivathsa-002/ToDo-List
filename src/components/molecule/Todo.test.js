import React from "react";
import {render,fireEvent} from "@testing-library/react";
import Todo from "./Todo";
import "@testing-library/jest-dom/extend-expect";
import {create} from "react-test-renderer";

describe("Todo testing", () => {
    test("Matches the snapshot", () => {
      const todo="Go to Job";
      const input = create(<Todo todo={todo}/>);
      expect(input.toJSON()).toMatchSnapshot();
    });
});

it("renders correctly <Todo>", () => {
    const todo = "Go to Job";
    const{getByTestId} = render(<Todo todo={todo}/>)

    expect(getByTestId("todo")).toBeInTheDocument()
})