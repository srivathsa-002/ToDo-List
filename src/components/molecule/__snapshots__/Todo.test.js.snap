// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo testing Matches the snapshot 1`] = `
<section>
  <div
    className="option"
    data-testid="todo"
  >
    <div
      className="option-text"
    >
      <p
        style={
          Object {
            "margin": "0",
          }
        }
      />
    </div>
    <div
      className="option-materials"
    >
      <div
        aria-describedby={null}
        className=""
        data-testid="edit"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Edit ToDo"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM5.92 19H5v-.92l9.06-9.06.92.92L5.92 19zM20.71 5.63l-2.34-2.34c-.2-.2-.45-.29-.71-.29s-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        aria-describedby={null}
        className=""
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Mark as Complete"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div>
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root"
                data-testid="checkbox"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    aria-label="checkbox with default color"
                    checked={false}
                    className="PrivateSwitchBase-input-4"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </span>
            </div>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        aria-describedby={null}
        className=""
        data-testid="delete"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Delete Todo"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5l-1-1h-5l-1 1H5v2h14V4h-3.5z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</section>
`;
